framework/src/modules/nft/internal_method.ts

```diff
@@ -12,30 +12,54 @@
  * Removal or modification of this copyright notice is prohibited.
  */
 
+/* eslint-disable max-classes-per-file */
 import { codec } from '@liskhq/lisk-codec';
 import { BaseMethod } from '../base_method';
 import { NFTStore, NFTAttributes } from './stores/nft';
-import { InteroperabilityMethod, ModuleConfig, NFTMethod } from './types';
-import { MethodContext } from '../../state_machine';
+import { InteroperabilityMethod, ModuleConfig, NFTMethod, TokenMethod } from './types';
+import { ImmutableMethodContext, MethodContext } from '../../state_machine';
 import { TransferEvent } from './events/transfer';
 import { UserStore } from './stores/user';
-import { CROSS_CHAIN_COMMAND_NAME_TRANSFER, MODULE_NAME_NFT, NFT_NOT_LOCKED } from './constants';
+import {
+	CROSS_CHAIN_COMMAND_NAME_TRANSFER,
+	LENGTH_CHAIN_ID,
+	MAX_LENGTH_DATA,
+	MODULE_NAME_NFT,
+	NFT_NOT_LOCKED,
+	NftErrorEventResult,
+	NftEventResult,
+} from './constants';
 import { EscrowStore } from './stores/escrow';
 import { TransferCrossChainEvent } from './events/transfer_cross_chain';
 import { crossChainNFTTransferMessageParamsSchema } from './schemas';
 
+export class TransferVerifyError extends Error {
+	public code: NftErrorEventResult;
+
+	public constructor(message: string, code: NftErrorEventResult) {
+		super(message);
+		this.code = code;
+	}
+}
+
 export class InternalMethod extends BaseMethod {
 	private _config!: ModuleConfig;
-	private _method!: NFTMethod;
+	private _nftMethod!: NFTMethod;
 	private _interoperabilityMethod!: InteroperabilityMethod;
+	private _tokenMethod!: TokenMethod;
 
 	public init(config: ModuleConfig): void {
 		this._config = config;
 	}
 
-	public addDependencies(method: NFTMethod, interoperabilityMethod: InteroperabilityMethod) {
-		this._method = method;
+	public addDependencies(
+		nftMethod: NFTMethod,
+		interoperabilityMethod: InteroperabilityMethod,
+		tokenMethod: TokenMethod,
+	) {
+		this._nftMethod = nftMethod;
```

Comment:
not related to this PR, but this is circular dependency. it should be one directional

Reply:
To be addressed in #9030